draft-ietf-rats-eat.md

```diff
@@ -1198,104 +1141,40 @@ The string identifying the JSON-encoded token MUST be one of the following:
 
-The primary use for this is to facilitate the implementation of a small and secure attester, perhaps purely in hardware.
-This small, secure attester implements COSE signing and only a few claims, perhaps just UEID and hardware identification.
-It has inputs for digests of submodules, perhaps 32-byte hardware registers.
-Software running on the device constructs larger claim sets, perhaps very large, encodes them and digests them.
-The digests are written into the small secure attesters registers.
-The EAT produced by the small secure attester only contains the UEID, hardware identification and digests and is thus simple enough to be implemented in hardware.
-Probably, every data item in it is of fixed length.
+#### Nested-Token {#Nested-Token}
 
-The integrity protection for the larger claims sets will not be as secure as those originating in hardware block, but the key material and hardware-based claims will be.
-It is possible for the hardware to enforce hardware access control (memory protection)  on the digest registers so that some of the larger claims can be more secure.
-For example, one register may be writable only by the TEE, so the detached claims from the TEE will have TEE-level security.
+The Nested-Token option provides a means of representing claims from a submodule that has its own attesting environment,
+i.e., it has keys distinct from the attester producing the surrounding token. Claims are represented in a signed EAT token. Inclusion of a signed
+eat as a claim cryptographically binds the EAT to the surrounding token.
+If it was conveyed in parallel with the surrounding token, there would be no such binding and attackers could substitute a good attestation from another device for the attestation of an errant subsystem.
 
-The data type for this type of submodule MUST be an array
-It contains two data items, a hash algorithm identifier and a byte string containing the digest.
+A nested token need not use the same encoding as the enclosing token.
+This enables composite devices to be built without regards to the encoding used by components.
+Thus, a CBOR-encoded EAT can have a JSON-encoded EAT as a nested token and vice versa.
 
-The hash algorithm identifier is always from the COSE Algorithm registry, {{IANA.COSE.Algorithms}}.
-Either the integer or string identifier may be used.
-The hash algorithm identifier is never from the JOSE Algorithm registry.
+#### Detached Submodule Digest
 
-When decoding a CBOR format token, the detached digest type is distinguished from the other types by it being an array.
-In CBOR encoded tokens none of other submodule types are arrays.
+The Detached-Submodule-Digest option is similar to a Claims-Set submodule, except a digest of the Claims-Set is included in the claim with the Claims-Set contents conveyed separately. The separately conveyed Claims-Set is called a detached claims set. The input to the digest algorithm is the byte-string wrapped encoded form of the detached Claims-Set.
```

Comment:
This is missing the formatting for hash input for JSON-encoded tokens, but that was missing from before.

We can probably get away with not having the hash cover the hash algorithm because the whole token is signed.